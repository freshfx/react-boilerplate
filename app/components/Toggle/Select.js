import styled from '@emotion/styled'

const Select = styled.select`
  line-height: 1em;
  background-color: transparent;
  border-style: none;
`

export default Select
