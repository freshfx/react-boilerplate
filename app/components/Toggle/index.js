import React from 'react'
import PropTypes from 'prop-types'
import noop from 'lodash/noop'

import Select from './Select'
import ToggleOption from '../ToggleOption'

function Toggle(props) {
  let content = <option>--</option>

  // If we have items, render them
  if (props.values.length) {
    content = props.values.map(value =>
      <ToggleOption key={value} value={value} message={props.messages[value]} />)
  }

  return (
    <Select value={props.value} onChange={props.onToggle}>
      {content}
    </Select>
  )
}

Toggle.propTypes = {
  messages: PropTypes.object,
  onToggle: PropTypes.func,
  value: PropTypes.string,
  values: PropTypes.array
}

Toggle.defaultProps = {
  messages: {},
  onToggle: noop,
  value: '',
  values: []
}

export default Toggle
