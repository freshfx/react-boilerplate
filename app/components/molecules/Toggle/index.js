import React, {useMemo} from 'react'
import PropTypes from 'prop-types'
import noop from 'lodash/noop'

import ToggleOption from 'components/atoms/ToggleOption'

import Select from './Select'

const Toggle = props => {
  const renderOption = value => (
    <ToggleOption key={value} value={value} message={props.messages[value]} />
  )

  const content = useMemo(() => {
    // If we have items, render them
    if (props.values.length) {
      return props.values.map(renderOption)
    }
    return <option>--</option>
  }, [props.messages, props.values])

  return (
    <Select value={props.value} onChange={props.onToggle}>
      {content}
    </Select>
  )
}

Toggle.propTypes = {
  messages: PropTypes.object,
  onToggle: PropTypes.func,
  value: PropTypes.string,
  values: PropTypes.array
}

Toggle.defaultProps = {
  messages: {},
  onToggle: noop,
  value: '',
  values: []
}

export default Toggle
