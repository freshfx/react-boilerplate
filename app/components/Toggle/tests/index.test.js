import React from 'react'
import {shallow} from 'enzyme'
import {IntlProvider, defineMessages} from 'react-intl'

import Toggle from '../index'

const zero = 0
const one = 1

describe('<Toggle />', () => {
  it('should contain default text', () => {
    const defaultEnMessage = 'someContent'
    const defaultDeMessage = 'someOtherContent'
    const messages = defineMessages({
      de: {
        defaultMessage: defaultDeMessage,
        id: 'boilerplate.containers.LocaleToggle.en'
      },
      en: {
        defaultMessage: defaultEnMessage,
        id: 'boilerplate.containers.LocaleToggle.en'
      }
    })
    const renderedComponent = shallow(<IntlProvider locale="en">
      <Toggle
        values={[
          'en',
          'de'
        ]} messages={messages} />
    </IntlProvider>)
    expect(renderedComponent.contains(<Toggle
      values={[
        'en',
        'de'
      ]} messages={messages} />)).toBe(true)
    expect(renderedComponent.find('option').length).toBe(zero)
  })
  it('should not have ToggleOptions if props.values is not defined', () => {
    const renderedComponent = shallow(<Toggle />)
    expect(renderedComponent.contains(<option>--</option>)).toBe(true)
    expect(renderedComponent.find('option').length).toBe(one)
  })
})
