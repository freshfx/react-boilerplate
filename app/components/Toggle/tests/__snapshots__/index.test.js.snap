// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle should match snapshot 1`] = `
.emotion-0 {
  line-height: 1em;
  height: 20px;
}

<select
  class="emotion-0 emotion-1"
>
  <option>
    --
  </option>
</select>
`;
